### PR TITLE
Burdened and Honorbound are now traumas, not mutations.

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

It was a terrible idea to make them mutations. You'll notice minute differences, maybe a different background on the honorbound declaration of guilt, but the biggest one is this:

## Why It's Good For The Game: PLASMAMEN CAN BE BURDENED AND HONORBOUND

Yep, that's right, because they didn't have genes they were not allowed to be these things. And all for entirely code reasons! Stupid stupid stupid.

## Changelog

:cl:
expansion: Burdened and Honorbound can now be chosen by plasmamen. Sorry boys!
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
okay but won't plasmamen still be unable to get any of the positive or negative mutations from the burdened sect?

**Comment by tralezab** [comment]
> okay but won't plasmamen still be unable to get any of the positive or negative mutations from the burdened sect?

true, but better than nothing.

File: code/modules/religion/burdened/burdened_trauma.dm [hunk 1]
```diff
@@ -52,7 +49,10 @@
  * Arguments:
  * * increase: whether to tick burden_level up or down 1
  */
-/datum/mutation/human/burdened/proc/update_burden(increase)
+/datum/brain_trauma/special/burdened/proc/update_burden(increase)
+	var/datum/dna/dna = owner?.dna
+	if(!dna)
+		qdel(src)
```
Comment (MrMelbert) [inline code/modules/religion/burdened/burdened_trauma.dm:55]:
```suggestion
	var/datum/dna/dna = owner?.dna
	if(!dna)
		qdel(src)
		return
```
Should probably quit out on the proc if you're qdeling yourself

File: code/modules/religion/burdened/burdened_trauma.dm [hunk 2]
```diff
@@ -44,6 +40,7 @@
 		COMSIG_CARBON_GAIN_TRAUMA,
 		COMSIG_CARBON_LOSE_TRAUMA,
 		))
+	..()
```
Comment (MrMelbert) [inline code/modules/religion/burdened/burdened_trauma.dm:43]:
```suggestion
	return ..()
```

File: code/modules/religion/burdened/burdened_trauma.dm [hunk 3]
```diff
@@ -1,20 +1,17 @@
...
+	lose_text = "<span class='warning'>You no longer feel the need to burden yourself!</span>"
+	random_gain = FALSE
 	/// goes from 0 to 6 (but can be beyond 6, just does nothing) and gives rewards. increased by disabling yourself with debuffs
 	var/burden_level = 0
 
-/datum/mutation/human/burdened/on_acquiring(mob/living/carbon/human/owner)
-	if(..())
-		return
+/datum/brain_trauma/special/burdened/on_gain()
+	..()
```
Comment (MrMelbert) [inline code/modules/religion/burdened/burdened_trauma.dm:14]:
```suggestion
	. = ..()
```

File: code/modules/religion/burdened/burdened_trauma.dm [hunk 4]
```diff
@@ -1,20 +1,17 @@
...
-	text_gain_indication = "<span class='notice'>You feel burdened!</span>"
-	text_lose_indication = "<span class='warning'>You no longer feel the need to burden yourself!</span>"
+///Burdened grants some mutations upon injuring yourself sufficiently
+
+/datum/brain_trauma/special/burdened
+	name = "Flagellating Compulsions"
+	desc = "Patient feels compelled to injure themselves in various incapacitating and horrific ways. There seems to be an odd genetic... trigger, following these compulsions may lead to?"
+	scan_desc = "damaged frontal lobe"
+	gain_text = "<span class='notice'>You feel burdened!</span>"
+	lose_text = "<span class='warning'>You no longer feel the need to burden yourself!</span>"
```
Comment (MrMelbert) [inline code/modules/religion/burdened/burdened_trauma.dm:8]: i think span macros work in this if you care, i don't really (up to you)

File: code/modules/religion/honorbound/honorbound.dm [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+	name = "Join Crusade"
+	desc = "Converts someone to your sect. They must be willing, so the first invocation will instead prompt them to join. \
+	They will become honorbound like you, and you will gain a massive favor boost!"
+	ritual_length = 30 SECONDS
+	ritual_invocations = list(
+	"A good, honorable crusade against evil is required.",
+	"We need the righteous ...",
+	"... the unflinching ...",
+	"... and the just.",
+	"Sinners must be silenced ...",)
```
Comment (MrMelbert) [inline code/modules/religion/honorbound/honorbound.dm:44]:
```suggestion
	ritual_invocations = list(
		"A good, honorable crusade against evil is required.",
		"We need the righteous ...",
		"... the unflinching ...",
		"... and the just.",
		"Sinners must be silenced ...",
	)
```

File: code/modules/religion/honorbound/honorbound_trauma.dm [hunk 6]
```diff
@@ -1,19 +1,17 @@
...
-	power = /obj/effect/proc_holder/spell/pointed/declare_evil
-	locked = TRUE
-	text_gain_indication = "<span class='notice'>You feel honorbound!</span>"
-	text_lose_indication = "<span class='warning'>You feel unshackled from your code of honor!</span>"
+/datum/brain_trauma/special/honorbound
+	name = "Dogmatic Compulsions"
+	desc = "Patient feels compelled to follow supposed \"rules of combat\"."
+	scan_desc = "damaged frontal lobe"
+	gain_text = "<span class='notice'>You feel honorbound!</span>"
+	lose_text = "<span class='warning'>You feel unshackled from your code of honor!</span>"
```
Comment (MrMelbert) [inline code/modules/religion/honorbound/honorbound_trauma.dm:8]: same here re: macros

File: code/modules/religion/honorbound/honorbound_trauma.dm [hunk 7]
```diff
@@ -32,7 +30,11 @@
 	//signal that checks for dishonorable attacks
 	RegisterSignal(owner, COMSIG_MOB_CLICKON, .proc/attack_honor)
 
-/datum/mutation/human/honorbound/on_losing(mob/living/carbon/human/owner)
+	//spell
```
Comment (MrMelbert) [inline code/modules/religion/honorbound/honorbound_trauma.dm:33]: Nice comment

File: code/modules/religion/honorbound/honorbound_trauma.dm [hunk 8]
```diff
@@ -1,19 +1,17 @@
...
-	text_gain_indication = "<span class='notice'>You feel honorbound!</span>"
-	text_lose_indication = "<span class='warning'>You feel unshackled from your code of honor!</span>"
+/datum/brain_trauma/special/honorbound
+	name = "Dogmatic Compulsions"
+	desc = "Patient feels compelled to follow supposed \"rules of combat\"."
+	scan_desc = "damaged frontal lobe"
+	gain_text = "<span class='notice'>You feel honorbound!</span>"
+	lose_text = "<span class='warning'>You feel unshackled from your code of honor!</span>"
+	/// the ability to declare others as evil!
+	var/obj/effect/proc_holder/spell/pointed/declare_evil/declare = /obj/effect/proc_holder/spell/pointed/declare_evil
```
Comment (MrMelbert) [inline code/modules/religion/honorbound/honorbound_trauma.dm:10]: Why do you even hold this reference? cause you don't handle the reference in `on_loss` or `destroy`
It doesn't seem necessary at all

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/religion/honorbound/honorbound_trauma.dm (hunk 7 above)
Comment (tralezab) [reply code/modules/religion/honorbound/honorbound_trauma.dm:33]: spell

**Comment by Coastfront** [comment]
Finally, plasmamen oppresion from gods has ended. Now plasmamen get rewarded for all their mutilation by having their god do absolutely fucking nothing for them, instead of just not letting them in the gates at all.

**Pull request closed without merging**